File: src/main/kotlin/de/nykon/productivity/value/Project.kt
package de.nykon.productivity.value

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime
import java.util.*

/**
 * Encapsulates multiple tasks and manages related users.
 */
@Document(collection = "projects")
data class Project(
    @Id val projectId: String = UUID.randomUUID().toString(),
    val name: String = "New project",
    val description: String? = null,
    @Indexed val email: String? = null,
    val userIds: List<String> = emptyList(),
    val createTime: LocalDateTime = LocalDateTime.now()
)

File: frontend/productivity/src/Project.js
class Project {
	constructor() {
        this.projectId = undefined;
        this.name = undefined;
        this.description = undefined;
        this.email = undefined;
        this.isProtected = false;
        this.ui = {
          xposition: 0,
          yposition: 0
        };
    }

	static from(json){
		return Object.assign(new Project(), json);
	  }
}

export default Project;